src/robot_state_publisher.cpp

```diff
@@ -332,64 +355,58 @@ rcl_interfaces::msg::SetParametersResult RobotStatePublisher::parameterUpdate(
 
   for (const rclcpp::Parameter & parameter : parameters) {
     if (parameter.get_name() == "robot_description") {
-      // First make sure that it is still a string
-      if (parameter.get_type() != rclcpp::ParameterType::PARAMETER_STRING) {
-        result.successful = false;
-        result.reason = "URDF must be a string";
-        break;
-      }
-
-      // Now get the parameter
       std::string new_urdf = parameter.as_string();
-      // And ensure that it isn't empty
+      // Ensure that it isn't empty
       if (new_urdf.empty()) {
         result.successful = false;
         result.reason = "Empty URDF is not allowed";
         break;
       }
 
+      // And that we can successfully parse it
       try {
-        setupURDF(new_urdf);
+        parseURDF(new_urdf);
       } catch (const std::runtime_error & err) {
         RCLCPP_WARN(get_logger(), "%s", err.what());
         result.successful = false;
         result.reason = err.what();
         break;
       }
-    } else if (parameter.get_name() == "frame_prefix") {
-      if (parameter.get_type() != rclcpp::ParameterType::PARAMETER_STRING) {
-        result.successful = false;
-        result.reason = "frame_prefix must be a string";
-        break;
-      }
-      frame_prefix_ = parameter.as_string();
-    } else if (parameter.get_name() == "ignore_timestamp") {
-      if (parameter.get_type() != rclcpp::ParameterType::PARAMETER_BOOL) {
-        result.successful = false;
-        result.reason = "ignore_timestamp must be a boolean";
-        break;
-      }
```

Comment:
Ok with the type checks, but we are removing [`ignore_timestamp_`](https://github.com/ros/robot_state_publisher/pull/180/files#diff-56a859e14fb1c861d179f60566024cf100f2a34b9c5d90fe1d727a04de5dd9f5L379) and [`frame_prefix_`](https://github.com/ros/robot_state_publisher/pull/180/files#diff-56a859e14fb1c861d179f60566024cf100f2a34b9c5d90fe1d727a04de5dd9f5L372) updates as well.

Reply:
Oh, right, that's a really good point.  Thanks for pointing it out.

In point of fact, we don't really need to store `ignore_timestamp_` and `frame_prefix_` inside of the class; it is already being stored in the parameter database, so we can always just fetch it from there.  I'll go ahead and implement it that way instead so that we don't have to have the member variables.

Reply:
Fixed in adac2b7